macros/edr/data_monitoring/monitors_query/table_monitoring_query.sql

```diff
@@ -1,78 +1,33 @@
-{% macro table_monitoring_query(monitored_table_relation, timestamp_column, min_bucket_start, table_monitors, freshness_column, where_expression, time_bucket) %}
+{% macro table_monitoring_query(monitored_table_relation, timestamp_column, min_bucket_start, table_monitors, where_expression, time_bucket, metric_args) %}
```

Comment:
The main changes are in this file:

1. I refactored the function a bit, so each metric implementation is in its own macro below - I felt it's needed as I added more logic here (in the beginning I thought to do an even bigger refactor but decided to wait with it...)
2. There are 3 metrics defined below - `row_count`, `freshness` (modified metric), and `event_freshness` (a new metric)

Reply:
Thanks, this is a great idea.
I thought about doing a larger refactor here but decided to hold off to another PR (and also break down the other anomaly queries while we're at it)

One thing that was in my mind though is that we may want at some point to be able to calculate metrics regardless of the tests being defined (e.g. for a data quality dashboard).